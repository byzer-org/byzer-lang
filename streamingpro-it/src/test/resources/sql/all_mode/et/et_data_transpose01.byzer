set jsonStr='''
    {"subject":"数学","name":"张三","score":88},
    {"subject":"语文","name":"张三","score":92}
    {"subject":"英语","name":"张三","score":77}
    {"subject":"数学","name":"王五","score":65}
    {"subject":"语文","name":"王五","score":87}
    {"subject":"英语","name":"王五","score":90}
    {"subject":"数学","name":"李雷","score":67}
    {"subject":"语文","name":"李雷","score":33}
    {"subject":"英语","name":"李雷","score":24}
    {"subject":"数学","name":"宫九","score":77}
    {"subject":"语文","name":"宫九","score":87}
    {"subject":"英语","name":"宫九","score":90}
''';

load jsonStr.`jsonStr` as data;
run data as DataTranspose.`` where method='pivot'
and indexCols='name'
and inputCol='subject'
and aggCol='score'
as data1;

-- try to unpivot
run data1 as DataTranspose.`` where method='unpivot'
and indexCols='name'
and mergedCol='sbject'
and unPivotCols='数学,英语,语文'
and aggCol='score'
as res;