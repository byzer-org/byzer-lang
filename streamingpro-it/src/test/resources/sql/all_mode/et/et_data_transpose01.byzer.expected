"name","sbject","score"
"王五","数学",65
"王五","英语",90
"王五","语文",87
"李雷","数学",67
"李雷","英语",24
"李雷","语文",33
"宫九","数学",77
"宫九","英语",90
"宫九","语文",87
"张三","数学",88
"张三","英语",77
"张三","语文",92